firebaseFunctions/src/index.ts

```diff
@@ -231,3 +197,143 @@ export const createDeal = functions.https.onRequest(
       }
     }),
 );
+
+/**
+ * Creates/updates a document in the users collection,
+ * which contains date when the request was made and returns it
+ */
+export const getDateToSign = functions.https.onRequest(
+  (request, response) =>
+  // Allow cross-origin requests for this function
+    cors(request, response, async () => {
+      if (request.method !== "POST") {
+        return response.sendStatus(403);
+      }
+
+      if (!request.body.address) {
+        return response.sendStatus(400);
+      }
+
+      /**
+       * Address coming from the request body
+       */
+      const address: string = request.body.address;
+
+      // Fail if provided address is not an ethereum address
+      try {
+        getAddress(address);
+      } catch {
+        return response.sendStatus(500);
+      }
+
+      try {
+        /**
+         * Current date as a UTC string, later used as a part of message the user is going to sign
+         */
+        const authenticationRequestDate = new Date().toUTCString();
+
+        // Create/update document for the provided address, storing the current time, to be used to create signature
+        await admin
+          .firestore()
+          .collection(USERS_COLLECTION)
+          .doc(address)
+          .set({authenticationRequestDate});
+
+        return response.status(200).json({ authenticationRequestDate });
+      } catch (error) {
+        functions.logger.error(error);
+        return response.sendStatus(500);
+      }
+    }),
+);
+
+/**
+ * Verifies that a message was signed by the provided address.
+ * Therefore it proofs the ownership of the address.
+ */
+export const verifySignedMessageAndCreateCustomToken = functions.https.onRequest(
+  (request, response) =>
+  // Allow cross-origin requests for this function
+    cors(request, response, async () => {
+      if (request.method !== "POST") {
+        return response.sendStatus(403);
+      }
+
+      if (!request.body.address || !request.body.signature) {
+        return response.sendStatus(400);
+      }
+
+      /**
+       * Address coming from the request body
+       */
+      const address: string = request.body.address;
+
+      // Fail if provided address is not an ethereum address
+      try {
+        getAddress(address);
+      } catch {
+        return response.sendStatus(500);
+      }
+
+      /**
+       * Signature of the message which should be signed by the address, coming from the request body
+       */
+      const signature: string = request.body.signature;
```

Comment:
"should"? 
The method comment explains
" * Verifies that a message was signed by the provided address."
So the message "was" already signed when we reached this piece of code?

As I see it, we could remove this comment, because code explains it already.